src/main/java/org/medicmobile/webapp/mobile/RequestLocationPermissionActivity.java

```diff
@@ -0,0 +1,124 @@
+package org.medicmobile.webapp.mobile;
+
+import static android.Manifest.permission.ACCESS_COARSE_LOCATION;
+import static android.Manifest.permission.ACCESS_FINE_LOCATION;
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+import static org.medicmobile.webapp.mobile.MedicLog.trace;
+
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Bundle;
+import android.provider.Settings;
+import android.view.View;
+import android.view.Window;
+import android.widget.TextView;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.core.content.ContextCompat;
+import androidx.fragment.app.FragmentActivity;
+
+/**
+ * Shows a confirmation view that displays a "prominent" disclosure about how
+ * the user geolocation data is used, asking to confirm whether to allow the app to
+ * access the location or not.
+ *
+ * If the user accepts, a request to the API to access the location is made by the main activity,
+ * but Android will show another confirmation dialog. If the user decline the first
+ * confirmation, the request to the API is omitted and the decision recorded to avoid
+ * requesting the same next time.
+ */
+public class RequestLocationPermissionActivity extends FragmentActivity {
+	/**
+	* TRIGGER_CLASS {String} Extra in the request intent to specify the class that trigger this activity,
+	*                        it will be passed on to the result intent. It can be used to continue the
+	*                        action of the trigger class after the intent is resolved.
+	*/
+	static final String TRIGGER_CLASS = "TRIGGER_CLASS";
+	static final String[] LOCATION_PERMISSIONS = { ACCESS_FINE_LOCATION, ACCESS_COARSE_LOCATION };
+
+	private ActivityResultLauncher<String[]> requestPermissionLauncher =
+		registerForActivityResult(new ActivityResultContracts.RequestMultiplePermissions(), grantedMap -> {
+			boolean allGranted = grantedMap.size() > 0 && !grantedMap.containsValue(false);
+			Intent responseIntent = createResponseIntent();
+
+			if (allGranted) {
+				trace(this, "RequestLocationPermissionActivity :: User allowed location permission.");
+				setResult(RESULT_OK, responseIntent);
+				finish();
+				return;
+			}
+
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+				boolean rationalFineLocation = !shouldShowRequestPermissionRationale(ACCESS_FINE_LOCATION);
+				boolean rationalCoarseLocation = !shouldShowRequestPermissionRationale(ACCESS_COARSE_LOCATION);
+
+				if (rationalFineLocation || rationalCoarseLocation) {
```

Comment:
Oh sorry I forgot to mention that we have a special code for that thing 😩
Currently, when the user denies the permission, we save that decision in the [SettingsStore](https://github.com/medic/cht-android/blob/e107f89f9873a4fbed52d83e485660c42f1fe6d1/src/main/java/org/medicmobile/webapp/mobile/SettingsStore.java#L84) to prevent asking again, it isn't a very good practice since we're not considering the case when the user rethinks the decision and wants to grant it later. I commented this code to test this escenario 😅.
We decided to remove the SettingsStore flag in in this [ticket](https://github.com/medic/cht-android/issues/207#issuecomment-1078722873).
Do you think we should remove that flag in this PR? 🤔

Reply:
Ah, got it!  Thanks for the explanation. Regarding removing it from the SettingsStore in this ticket, I think it probably comes down to if we want to be able to AT this behavior.  

Since we have unit tests for it and we will be able to AT the behavior of the matching storage permission, I am comfortable merging this as-is. But, I would not object if you wanted to also include the SettingsStore changes in this PR or open a separate issue to add this functionality after the #270 is addressed.

Reply:
Thanks for the insight! I'm going to remove the flag in SettingsStore in the other ticket.